x/operator/keeper/operator.go

```diff
@@ -54,7 +54,30 @@ func (k *Keeper) IsOperator(ctx sdk.Context, addr sdk.AccAddress) bool {
 	return store.Has(addr)
 }
 
-func (k *Keeper) UpdateOptedInfo(ctx sdk.Context, operatorAddr, avsAddr string, info *operatortypes.OptedInfo) error {
+func (k *Keeper) HandleOptedInfo(ctx sdk.Context, operatorAddr, avsAddr string, handleFunc func(info *operatortypes.OptedInfo)) error {
+	opAccAddr, err := sdk.AccAddressFromBech32(operatorAddr)
+	if err != nil {
+		return errorsmod.Wrap(err, "HandleOptedInfo: error occurred when parse acc address from Bech32")
+	}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorOptedAVSInfo)
+	infoKey := assetstype.GetJoinedStoreKey(operatorAddr, avsAddr)
+	ifExist := store.Has(infoKey)
+	if !ifExist {
+		return errorsmod.Wrap(operatortypes.ErrNoKeyInTheStore, fmt.Sprintf("HandleOptedInfo: key is %suite", opAccAddr))
+	}
+	// get info from the store
+	value := store.Get(infoKey)
+	info := &operatortypes.OptedInfo{}
+	k.cdc.MustUnmarshal(value, info)
+	// call the handleFunc
+	handleFunc(info)
+	// restore the info after handling
+	bz := k.cdc.MustMarshal(info)
+	store.Set(infoKey, bz)
+	return nil
+}
+
+func (k *Keeper) SetOptedInfo(ctx sdk.Context, operatorAddr, avsAddr string, info *operatortypes.OptedInfo) error {
```

Comment:
In the SetOptedInfo function, in addition to verifying the address format,  is it necessary to verify that the **info** parameter contains all the necessary information required to join the AVS and that this information conforms to the predefined rules and constraints ？

Reply:
and also  SetOperatorInfo function.

Reply:
It seems we don't need to consider this for now. Both of these interfaces are modifications to the KV store at the low level. If we need to verify the information, it might be best to determine which parameters need checking and perform the parameter validation at the entry point of the call.